tmp/build/patch-charts.sh

```diff
@@ -252,6 +252,13 @@ function patchTemplates() {
     sed -i -e 's/\/health/\/tmp\/health/' ${HELM_DIR}/istio/charts/sidecarInjectorWebhook/templates/deployment.yaml
   fi
 
+  # add oauth-proxy default hub, tag and imagePullPolicy
```

Comment:
I think we need to distinguish between community and product.  Sadly, it seems there are two versions for product, openshift3/oauth-proxy and openshift4/ose-oauth-proxy, along with version specific tags (e.g. 3.11, 4.0, 4.1, etc.).  There also appears to be an openshift/origin-oauth-proxy on quay.io, which is specific to 4.x.

I wonder if there is some metadata we can pull from openshift to determine some of these settings (e.g. namespace [openshift, openshift3, openshift4], name prefix ["", ose, origin], tag [3.0, 4.1, etc.]?

Reply:
So, currently we are using the hard coded value openshift/oauth-proxy (always community). With this change we give the users the option to override that value, which is better than we currently have.

I agree that having better defaults is better. Since we are not officially supporting OCP 3, I think we can go with product for 4. Users still have the power to override it if they want.